primitives/src/pcs/univariate_kzg/mod.rs

```diff
@@ -652,6 +651,255 @@ where
     }
 }
 
+#[cfg(feature = "icicle")]
+pub(crate) mod icicle {
+    use super::*;
+    use crate::icicle_deps::{curves::*, *};
+    use ark_ec::{short_weierstrass::Affine, CurveConfig};
+    use ark_ff::BigInteger;
+
+    /// Trait for GPU-accelerated PCS.commit APIs
+    pub trait GPUCommit<E, C>
+    where
+        C: IcicleCurve + MSM<C>,
+        C::ScalarField: ArkConvertible<ArkEquivalent = E::ScalarField>,
+        C::BaseField: ArkConvertible<ArkEquivalent = <C::ArkSWConfig as CurveConfig>::BaseField>,
+        <C::ArkSWConfig as CurveConfig>::BaseField: PrimeField,
+        E: Pairing<G1Affine = Affine<<C as IcicleCurve>::ArkSWConfig>>,
+    {
+        /// The full cycle of computing poly-commit on GPU
+        fn commit_with_gpu(
```

Comment:
Should we add another function, which assumes that the srs is already on GPU and takes polys as input.

Reply:
this function accepts `HostOrDevice`, so srs and poly can be any one of the 4 combination states. this function shouldn't fail.

if either srs or poly are not on device, the `msm()` from ICICLE side would push it to GPU first, else it will just use the device ref we pass in.

so I'd say we don't need this another function imo

Reply:
I mean a full cycle in VID would be, a pre-loaded SRS is already on GPU, and some interface should takes (a list of) polys and return a commitment

Reply:
Current APIs are already sufficient.

I'm overthinking that, should we have a PCS struct, whose `ProverParam` is a `HostOrDeviceSlice`. We don't have to do it now.

Reply:
i don't think that's a good idea, the main concern is icicle is undergoing major breaking changes https://github.com/ingonyama-zk/icicle/pull/412, and I would expect significant changes, so fewer places we use them for now, the easier later when we upgrade.